modulo5/LabECommerce/src/endpoints/RegisterPurchases.ts

```diff
@@ -0,0 +1,69 @@
+import { Request, Response } from "express";
+import connection from "../database/connection";
+import { Purchases } from "../types/types";
+
+
+
+export const RegisterPurchases = async (req: Request, res: Response): Promise<void> => {
+    try {
+        let statusCode = 400
+        const { user_id, product_id, quantity } = req.body;
+
+        const checkUser = await connection.raw(`
+        SELECT * FROM labecommerce_users
+        WHERE id = "${user_id}"
+        `)
+
+
+        if (!user_id|| !product_id || !quantity) {
+            statusCode = 422
+            throw new Error("Parâmetro 'user_id', 'product_id' ou 'quantity' não informado.");
+        }
+
+        if(typeof user_id && typeof product_id !== "string"){
```

Comment:
Da forma que estava antes voce estava validando o campo `product_id` 

```suggestion
        if(typeof user_id  !== "string" && typeof product_id !== "string"){
```